### PR TITLE
Issue 741: Switch to Spotbugs

### PR DESCRIPTION
**Pull request by mino181295**
Moved from findbugs plugin to spotbugs.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -197,9 +197,9 @@
...
     <plugins>
       <plugin>
-        <groupId>org.codehaus.mojo</groupId>
-        <artifactId>findbugs-maven-plugin</artifactId>
-        <version>${findbugs-maven-plugin.version}</version>
+        <groupId>com.github.spotbugs</groupId>
+        <artifactId>spotbugs-maven-plugin</artifactId>
+        <version>${spotbugs-maven-plugin.version}</version>
         <configuration>
           <excludeFilterFile>${session.executionRootDirectory}/buildtools/src/main/resources/bookkeeper/findbugsExclude.xml</excludeFilterFile>
```
Comment (ivankelly) [inline pom.xml]: Should probably rename this file also

Comment (eolivelli) [reply pom.xml]: @ivankelly there is already an issue, I have already sent emails in the past months about this change.
I think we change keep findbugsExclude.xml as effectively SpotBugs is Findbugs (it just changed owner) and internally the format of the file is the same. The SportBugs maven plugin is a port of the Findbugs Maven plugin and all the configs are the same.

But if you feel strong about this we can change it as well.

**Comment by sijie** [comment]
committers: when you merge this pull request, please update all the jenkins job with the new command

**Comment by eolivelli** [comment]
@sijie sorry, I missed your comment before sending my approval

**Comment by eolivelli** [comment]
@sijie I will take care of updating the jobs on Jenkins

File: pom.xml (hunk 1 above)
Comment (ivankelly) [reply pom.xml]: @eolivelli it still needs to be announced when done, so people know they need to run spotbugs:check instead of findbugs:check

Comment (ivankelly) [reply pom.xml]: filename change isn't important, i'd just update it if other things need an update

**Comment by eolivelli** [comment]
@sijie I think @mino181295 addressed all of your comments, I will wait you acknowledge

**Comment by sijie** [comment]
@eolivelli I merged this PR and updated CI jobs.

**Pull request closed without merging**